### DISCUSSION TITLE
How do register and handle a custom URI scheme/deeplink

### DISCUSSION THREAD

**Discussion by ctigrisht** [AvaloniaUI/Avalonia#13366] [Q&A]
Hey, so I want to handle a custom uri scheme, like `myapp://someaction`, but it seems I can't find anything online, is this feature even implemented? Thanks

**Comment by thevortexcloud** [comment]
Isn't this more an OS concern rather than anything to do with Avalonia? EG 

https://stackoverflow.com/questions/80650/how-do-i-register-a-custom-url-protocol-in-windows

**Reply by ctigrisht** [reply]
Sure, but an app framework could provide such an abstraction.

**Reply by thevortexcloud** [reply]
As far as I am aware Avalonia has no built in way to do this. But one of the core Avalonia devs may say otherwise. I personally would think it's kind of weird if it does. Avalonia is primarily a GUI framework. not a general app framework. 

You can probably write your own abstraction somewhat easily though for most platforms.

**Reply by ctigrisht** [reply]
I'll look into it then, or just start up a localhost server to handle requests

**Reply by thevortexcloud** [reply]
As far as I can tell on Windows you just need to add the registry keys and on Linux you need to make and edit some files and then register it.

https://stackoverflow.com/questions/32064229/custom-protocol-handlers-linux-centos-7-for-chrome

For all other platforms, it's probably something similar.
